### ISSUE TITLE
installing chocolatey via OneGet nukes C:\Chocolatey\bin

### ISSUE THREAD

**Issue by 0xabu** [OneGet/ChocolateyProvider#4]
Possibly related to #3...

I started with a clean install of Win10. I then installed a bunch of chocolatey packages through OneGet, using version 2.8.5.130 of the Chocolatey provider. These packages ended up in C:\Chocolatey\lib... with shims in C:\Chocolatey\bin. So far so good (although I had to manually add bin to my PATH).

I wanted to play with the choco command line, to be able to do upgrades etc., and made the mistake of using OneGet to do it via "Install-Package chocolatey", which pulled down chocolatey v0.9.9.8. This installed itself under C:\Chocolatey\lib\, but it didn't know about any of the previously-installed packages (they still showed up in OneGet's Get-Package, but not in choco list). Worse, the chocolatey install nuked all the shims other than the new choco.bat in C:\Chocolatey\bin, and after uninstalling the wonky Chocolatey package I now have an empty bin directory :(

**Comment by ferventcoder** [comment]
I'm not sure choco is the wonky one here. The chocolatey provider is still very much a preview from awhile ago. I would wait until there is an announcement that it is ready to go before attempting to use it.

**Comment by ferventcoder** [comment]
Choco 0.9.9.x+ will ensure that you are not using c:\chocolatey as it is world-writeable and very insecure. That is if the installer works appropriately.

**Comment by 0xabu** [comment]
@ferventcoder, I'm not trying to assign blame. I get that it's a preview, I just figured that others may run into the same problem -- it's very easy to install out of the box on Win10 ("Get-PackageProvider chocolatey" prompts you to do it) and there are a few blogs and howtos directing people to do just this. If this isn't the right place to report it, where is?

**Comment by ferventcoder** [comment]
This is fine.

**Comment by ferventcoder** [comment]
It may actually be better to report it over here - https://github.com/chocolatey/chocolatey-oneget

**Comment by ferventcoder** [comment]
I know folks are blogging about it as well, which is not helpful when it is still the CTP from April 2014 that @fearthecowboy put together. I think I'll probably just need to sit down with Garrett or somebody so we can get the updated version banged out and at least a newer CTP released.

**Comment by 0xabu** [comment]
Now I'm confused about https://github.com/chocolatey/chocolatey-oneget vs https://github.com/OneGet/ChocolateyProvider The code in the former looks stale compared to the latter (here). Are you sure you want the issues over there?

**Comment by ferventcoder** [comment]
You know, I don't know for sure. That's a great question for @Jaykul and @fearthecowboy. I guess I'm not really sure where the development is going to.

**Comment by Jaykul** [comment]
I don't even ... 

Whatever you do, don't file it against [OneGet/ChocolateyProvider](https://github.com/OneGet/ChocolateyProvider) because even though it doesn't look like it ... that's really just @fearthecowboy's personal copy of my [jaykul/ChocolateyProvider](https://github.com/jaykul/ChocolateyProvider/) repo ... which should be the same as [chocolatey/chocolatey-oneget](https://github.com/chocolatey/chocolatey-oneget) ... don't ask me how we could ever get github to understand the truth :eyes: 

Anyway, the code that actually got used is what's in  [OneGet/chocolateyprototype](https://github.com/OneGet/chocolateyprototype), I think -- but I think if there's a bug, it's the fact that the NewChocolatey installer nuked folders that already existed?

**Comment by ferventcoder** [comment]
@Jaykul Hmmm :)

**Comment by ferventcoder** [comment]
Looks like you forked off of this repo, not the chocolatey/chocolatey-oneget repo. Probably where the confusion in github lies. :/

**Comment by Jaykul** [comment]
Well, I did technically fork his repo to get the "SDK" ... before you even knew there was a project ;-)

Then later you created a repo (not forked off either) which I just added as a remote.  As far as I'm concerned, the real repo is on my dev box and all of these are just remotes I push to when things are happy :-P

**Comment by ferventcoder** [comment]
:)

**Comment by Jaykul** [comment]
But my point is that for the "new" provider, we need to treat chcolatey's copy as the official one. I could delete my repo and re-create it as a fork of yours if that would help ;)

But this bug isn't really in the provider, I don't think. The provider's just calling the installer..

**Comment by ferventcoder** [comment]
Agreed - the bug is really that the provider is using `c:\chocolatey`
